yt/data_objects/particle_trajectories.py

```diff
@@ -137,10 +139,18 @@ def __init__(self, outputs, indices, fields=None, suppress_logging=False):
 
     def has_key(self, key):
         return (key in self.field_data)
-    
+
     def keys(self):
         return self.field_data.keys()
 
+    def determine_field(self, field):
```

Comment:
maybe give this a different name that's more distinct from `_determine_fields`? If I wasn't reading closely I might miss the distinction.